Recipe validation error should contain recipe

## What problem are you trying to solve?
I'd like to understand which recipe actually caused which validation error. Sample output from the Maven plugin:
```
[INFO] Validating active recipes...
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in org.openrewrite.java.security.OwaspTopTen.recipeList[6] (in jar:file:/C:/Users/Admin/.m2/repository/org/openrewrite/recipe/rewrite-java-security/2.0.4/rewrite-java-security-2.0.4.jar!/META-INF/rewrite/owasp.yml): recipe 'org.openrewrite.java.security.OwaspA010' does not exist.
[ERROR] Recipe validation error in scope: is required
[ERROR] Recipe validation error in builderType: is required
[ERROR] Recipe validation error in builderType: must not be blank
[ERROR] Recipe validation error in cveList: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
[ERROR] Recipe validation error in runtime: is required
```
The problem stems from the validation just collecting a list of failure messages, without any relation to the recipe(s): https://github.com/openrewrite/rewrite/blob/7ef3ffcb9d659eb57437ad47dce3ed474b719117/rewrite-core/src/main/java/org/openrewrite/Recipe.java#L318

## Describe the solution you'd like
make validateAll(...) return a Map<Recipe, Collection<Validated<Object>>> instead, such that every build tool can show recipe and failure.

--- a/rewrite-core/src/main/java/org/openrewrite/Recipe.java
+++ b/rewrite-core/src/main/java/org/openrewrite/Recipe.java
@@ -299,12 +299,13 @@ public abstract class Recipe implements Cloneable {
      */
     public Validated<Object> validate() {
         Validated<Object> validated = Validated.none();
-        List<Field> requiredFields = NullUtils.findNonNullFields(this.getClass());
+        Class<? extends Recipe> clazz = this.getClass();
+        List<Field> requiredFields = NullUtils.findNonNullFields(clazz);
         for (Field field : requiredFields) {
             try {
-                validated = validated.and(Validated.required(field.getName(), field.get(this)));
+                validated = validated.and(Validated.required(clazz.getSimpleName() + '.' + field.getName(), field.get(this)));
             } catch (IllegalAccessException e) {
-                logger.warn("Unable to validate the field [{}] on the class [{}]", field.getName(), this.getClass().getName());
+                logger.warn("Unable to validate the field [{}] on the class [{}]", field.getName(), clazz.getName());
             }
         }
         for (Recipe recipe : getRecipeList()) {

